dbms/src/TiDB/Schema/TiDBSchemaSyncer.h

```diff
@@ -144,30 +154,59 @@ struct TiDBSchemaSyncer : public SchemaSyncer
         return it->second;
     }
 
-    bool tryLoadSchemaDiffs(Getter & getter, Int64 version, Context & context)
+    // Return Values
+    // - if latest schema diff is not empty, return the (latest_version)
+    // - if latest schema diff is empty, return the (latest_version - 1)
+    // - if error happend, return (-1)
+    Int64 tryLoadSchemaDiffs(Getter & getter, Int64 latest_version, Context & context)
     {
-        if (isTooOldSchema(cur_version, version))
+        if (isTooOldSchema(cur_version, latest_version))
         {
-            return false;
+            return -1;
         }
 
-        LOG_FMT_DEBUG(log, "try load schema diffs.");
+        LOG_FMT_DEBUG(log, "Try load schema diffs.");
 
-        SchemaBuilder<Getter, NameMapper> builder(getter, context, databases, version);
+        SchemaBuilder<Getter, NameMapper> builder(getter, context, databases, latest_version);
 
         Int64 used_version = cur_version;
-        std::vector<SchemaDiff> diffs;
-        while (used_version < version)
+        // First get all schema diff from `cur_version` to `latest_version`. Only apply the schema diff(s) if we fetch all
+        // schema diff without any exception.
+        std::vector<std::optional<SchemaDiff>> diffs;
+        while (used_version < latest_version)
         {
             used_version++;
             diffs.push_back(getter.getSchemaDiff(used_version));
         }
-        LOG_FMT_DEBUG(log, "end load schema diffs with total {} entries.", diffs.size());
+        LOG_FMT_DEBUG(log, "End load schema diffs with total {} entries.", diffs.size());
+
         try
         {
-            for (const auto & diff : diffs)
+            for (size_t diff_index = 0; diff_index < diffs.size(); diff_index++)
```

Comment:
```suggestion
            for (size_t diff_index = 0; diff_index < diffs.size(); ++diff_index)
```

Reply:
what for change `a++` to `++a`？

Reply:
`++a` is more effective. https://stackoverflow.com/questions/24886/is-there-a-performance-difference-between-i-and-i-in-c

Reply:
just dump, example here.

```
a++
mov eax, double word ptr [a]
add eax, 1
mov double word ptr [a], eax

++a
mov eax, double word ptr [a]
add eax, 1
mov double word ptr [a], eax
```
so only do `a++` or `++a` here, won't be any effective difference.

Reply:
yep, it will be optimize by compiler, there will not be any difference in binary, but give pressure to compiler. We can find difference by set `-O0` when compiling. Anyway, there are only tiny difference. The others LGTM.